src/org/folio/rest_v2/Permissions.groovy

```diff
@@ -166,4 +166,28 @@ class Permissions extends Authorization {
         logger.info("Wait a minute for permissions cache update")
         sleep(60000)
     }
+
+/**
+ * Adds a missing permission to a user.
+ * source: RANCHER-895
+ * @param tenant The tenant in which the user is located.
+ * @param user The user to whom the permission is added.
+ */
+    void addUserMissingPermission(OkapiTenant tenant, OkapiUser user) {
+        def allPermissions = getAllPermissions(tenant)
+        def existingPermissions = getUserPermissions(tenant, user)
+        allPermissions.each { permissionName ->
+            if ( existingPermissions.contains(permissionName) ) {
+                logger.info("User: ${user} already has permission: ${permissionName}")
+            }
+            else {
+                try {
+                    addUserPermission(tenant, user, permissionName as String)
+                }
+                catch (e){
+                    logger.warning("Add permission operation: ${permissionName} failed with error: ${e.getMessage()}")
+                }
+            }
+        }
+    }
```

Comment:
To avoid multiple calls to okapi, it is better to first generate a list of missing permissions and then add them in batch with addUserPermissions method.

Reply:
Hello Sir @OHaimanov 
Would you please guide me, how it could generated(meaning a list of missing permissions) I tried to search for it, was not able to find. Thanks!